tests/Functional/src/com/microsoft/aad/adal/test/BrokerProxyTests.java

```diff
@@ -211,6 +216,169 @@ public void testCanSwitchToBrokerValid_Skip() throws IllegalArgumentException,
         assertFalse("This should skip broker", result);
     }
 
+    @SmallTest
+    public void testVerifyManifestPermission_valid() throws NoSuchAlgorithmException, NoSuchPaddingException, IllegalArgumentException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException, NameNotFoundException, NoSuchFieldException {
+        //mock Brokerproxy
+        FileMockContext mockContext = new FileMockContext(getContext());
+        Boolean actual;
+        mockContext.setPermission("android.permission.INTERNET",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.GET_ACCOUNTS",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.MANAGE_ACCOUNTS",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.USE_CREDENTIALS",PackageManager.PERMISSION_GRANTED);
+        Object brokerProxy = ReflectionUtils.getInstance("com.microsoft.aad.adal.BrokerProxy");
+        String authenticatorType = AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE;
+        String brokerPackage = AuthenticationConstants.Broker.PACKAGE_NAME;
+        String contextPackage = "com.test";
+        Signature signature = new Signature(testSignature);
+        AuthenticationSettings.INSTANCE.setBrokerSignature(testTag);
+        Account[] accts = getAccountList("currentUserName", authenticatorType);
+        prepareProxyForTest(brokerProxy, authenticatorType, brokerPackage, contextPackage,
+                signature, true, accts);
+        ReflectionUtils.setFieldValue(brokerProxy, "mContext", mockContext);
+        
+        //set the access to private method
+        Method m = ReflectionUtils.getTestMethod(brokerProxy, "verifyManifestPermissions");
+        
+        //test@case all permissions are granted
+        actual = (Boolean)m.invoke(brokerProxy);
+        assertTrue(actual);
+    }
+    
+    public void testVerifyManifestPermission_missingPermission_MANAGE_ACCOUNTS() throws NoSuchAlgorithmException, NoSuchPaddingException, IllegalArgumentException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException, NameNotFoundException, NoSuchFieldException {
+        //mock Brokerproxy
+        FileMockContext mockContext = new FileMockContext(getContext());
+        mockContext.setPermission("android.permission.INTERNET",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.GET_ACCOUNTS",PackageManager.PERMISSION_GRANTED);
```

Comment:
ha, interesting. I looked at this code review on a laptop with a smaller screen. nvm, forget this comment. Sorry.